relay-sampling/src/evaluation.rs

```diff
@@ -195,32 +229,30 @@ impl<'a> SamplingEvaluator<'a> {
         I: Iterator<Item = &'b SamplingRule>,
     {
         for rule in rules {
-            if !rule.condition.matches(instance) {
+            if !rule.time_range.contains(self.now) || !rule.condition.matches(instance) {
```

Comment:
moving the time range constraint here makes no difference in the logic, it only improves performance. Previously it was done right after in SamplingRule::evaluate